llvm/lib/Analysis/ScalarEvolution.cpp

```diff
@@ -7299,6 +7309,11 @@ bool ScalarEvolution::isGuaranteedToTransferExecutionTo(const Instruction *A,
   return false;
 }
 
+bool ScalarEvolution::isGuaranteedNotToBePoison(const SCEV *Op) {
+  SCEVPoisonCollector PC(/* LookThroughMaybePoisonBlocking */ false);
```

Comment:
```suggestion
  SCEVPoisonCollector PC(/* LookThroughMaybePoisonBlocking */ true);
```
I think? If we have `%a umin_seq %b` where `%b` may be poison, the whole expression may be poison (if `%a != 0`).

Reply:
Yep, updated, thanks!